tests/commands/test_new.py

```diff
@@ -0,0 +1,236 @@
+            self.command_tester.execute([('target', 'new_project'), ('--repo', 'MasoniteFramework/unknown_repo')])
+
+        self.assertTrue(str(e.exception).startswith("Not Found(404)"))
+
+    @responses.activate
+    def test_private_repos_errors_are_handled(self):
+        responses.add(responses.GET,
+            'https://api.github.com/repos/MasoniteFramework/secret/releases',
+            status=403,
+        )
+
+        with self.assertRaises(ProjectProviderHttpError) as e:
+            self.command_tester.execute([('target', 'new_project'), ('--repo', 'MasoniteFramework/secret')])
+
+        self.assertTrue(str(e.exception).startswith("Forbidden(403)"))
+
+    @responses.activate
+    def test_timeouts_are_handled(self):
+        responses.add(responses.GET,
+            'https://api.github.com/repos/{0}/releases'.format(self.default_repo),
+            body=requests.Timeout(),
+        )
+
+        with self.assertRaises(ProjectProviderTimeout) as e:
+            self.command_tester.execute([('target', 'new_project')])
+
+        self.assertTrue(str(e.exception).startswith("github provider is not reachable"))
+
+    def test_download_errors_are_displayed(self):
+        pass
+
+    # Following tests are close to integration tests but still quick and without rate limiting issue so
+    # they can be ran as unit tests
+    # ----------------------------------------------------------------------------------------------
+    @responses.activate
+    def test_can_craft_default_repo_successfully(self):
+        # still mocking requests to avoid failures from api rate limits
+        responses.add(responses.GET,
+            'https://api.github.com/repos/{0}/releases'.format(self.default_repo),
+            body='[{"name": "v2.3.6", "tag_name": "v2.3.6", "zipball_url": "https://api.github.com/repos/MasoniteFramework/cookie-cutter/zipball/v2.3.6", "prerelease": false}]'
+        )
+        responses.add(responses.GET,
+            'https://api.github.com/repos/{0}/releases/tags/v2.3.6'.format(self.default_repo),
+            body='{"zipball_url": "https://api.github.com/repos/MasoniteFramework/cookie-cutter/zipball/v2.3.6"}'
+        )
+        # authorize requests not using the API
+        responses.add_passthru('https://api.github.com/repos/MasoniteFramework/cookie-cutter/zipball/v2.3.6')
+        responses.add_passthru('https://codeload.github.com/MasoniteFramework/cookie-cutter/legacy.zip/v2.3.6')
+
+        self.command_tester.execute([('target', 'new_project')])
+        self.assertTrue(
+            "Project Created Successfully" in self.command_tester.get_display()
+        )
+        # verify that project has really been created by checking files
+        self.assertTrue("craft" in os.listdir(self.test_project_dir))
+        self.assertTrue("app" in os.listdir(self.test_project_dir))
+
+    @responses.activate
+    def test_can_craft_default_repo_successfully_with_release(self):
```

Comment:
@girardinsamuel can you take a look at this? All these tests are failing

Reply:
Yes I am looking into it

Reply:
On the PR branch (feat/342), all tests are still passing in my local env 
![image](https://user-images.githubusercontent.com/9897999/100200975-b4376700-2eff-11eb-905f-9adb756aa60a.png)

I will check difference in requirements

Reply:
It's due to the upgrade in `cleo` version, I might have to edit the tests, but it should not be too complicated. I'll do this later today 😉

Reply:
@josephmancuso 
It's fixed 💪 . 
The API of cleo commands testers and the Command API changed a bit failing the tests. I had to looked up `cleo` code as the online doc has not been updated since 0.6.2..

I made a second commit to fix some other commands which are not tested thus not failing but which should fail now because of the change in Command cleo API: `self.error` => `self.line_error`.
https://github.com/MasoniteFramework/masonite/pull/367